### DISCUSSION TITLE
Bug:  (Failed to download Chrome for Linux from https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/linux64/chrome-linux64.zip).

### DISCUSSION THREAD

**Discussion by romarioputra** [hardkoded/puppeteer-sharp#2457] [Q&A]
### Description

[Description of the issue]
I use the default example of PuppeterSharp for **Generate PDF Files** and i got an error like title said.
When I look into the code, my code fails when using await browserFetcher.DownloadAsync();
Specifically when the code calls await _customFileDownload
### Complete minimal example reproducing the issue
I already put my api in Azure Functions Linux https://htmltopdffunclinux.azurewebsites.net/api/converted
Just call an api with URL as a request body
You can use postman for this

E.g.

```csharp
        private async Task<byte[]> ConvertHtmlToPdf(string url)
        {
            try
            {
                using var browserFetcher = new BrowserFetcher();
                await browserFetcher.DownloadAsync();
                // from this code, i get the error failed to download chrome for linux
                await using var browser = await Puppeteer.LaunchAsync(new LaunchOptions { Headless = true });
                await using var page = await browser.NewPageAsync();
                await page.GoToAsync(url);
                await page.EvaluateExpressionHandleAsync("document.fonts.ready");
                var bytes = await page.PdfDataAsync();
                return bytes;
            }
            catch (Exception)
            {
                throw;
            }
        }
```

// Act
...

// Assert
...
```

### Expected behavior:
download pdf

[What you expect to happen]

### Actual behavior:

[What actually happens]
(Failed to download Chrome for Linux from https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/linux64/chrome-linux64.zip).
### Versions

* Which version of PuppeteerSharp are you using? 13.0.2
* Which .NET runtime and version are you targeting? .NET CORE 7

**Comment by kblok** [comment]
`https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/linux64/chrome-linux64.zip` is a valid url. Did you check the inner exception?

**Comment by romarioputra** [comment]
> `https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/linux64/chrome-linux64.zip` is a valid url. Did you check the inner exception?

Do you mean the **ex** in throw new error(failed to download chrome for linux, **ex**)?

**Comment by kblok** [comment]
> > `https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/linux64/chrome-linux64.zip` is a valid url. Did you check the inner exception?
> 
> 
> 
> Do you mean the **ex** in throw new error(failed to download chrome for linux, **ex**)? 

Yes. Can you share the inner exception and the stack trace?

**Comment by romarioputra** [comment]
> > > `https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/linux64/chrome-linux64.zip` is a valid url. Did you check the inner exception?
> > 
> > 
> > Do you mean the **ex** in throw new error(failed to download chrome for linux, **ex**)?
> 
> Yes. Can you share the inner exception and the stack trace?

Here is my inner exception

{ "Value": "One or more errors occurred. (An exception occurred during a WebClient request.)", "Formatters": [], "ContentTypes": [], "DeclaredType": null, "StatusCode": 400 }

**Comment by kblok** [comment]
If you want to run that in azure functions you will have to use a docker image. You won't want to download the browser on every run.

This is quite an old post but it should help you get started https://www.hardkoded.com/blog/running-puppeteer-sharp-azure-functions

**Comment by ianpowell2017** [comment]
I had the same problem and it was because it couldn't write the downloaded file to the download location.
Try setting the `CacheDir` folder to a writeable location.  Here's an example.
```
using var browserFetcher = new BrowserFetcher
{
    CacheDir = "/tmp"
};
var download = await browserFetcher.DownloadAsync(BrowserTag.Stable);
```
You will need to read the download.GetExecutablePath() to find the chrome driver has been downloaded to.

**Comment by jonasivy** [comment]
Having the same issue but at GH workflow.  Have anyone else resolved this issue?
